components/CollectionForm.js

```diff
@@ -0,0 +1,115 @@
+import { useState } from "react";
+import styled from "styled-components";
+import Button from "@/components/Button";
+
+const StyledFormContainer = styled.div`
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+  width: 100vw;
+  padding: 3em;
+`;
+
+const StyledForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  width: 250px;
+  gap: 10px;
+`;
+
+const StyledColors = styled.button`
+  width: 24px;
+  height: 24px;
+  border-radius: 50px;
+  border: none;
+  cursor: pointer;
+  background-color: ${(prop) => prop.$inputColor};
+`;
+
+const StyledColorPicker = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  gap: 12px;
+`;
+
+const StyledButtonContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-evenly;
+  margin-top: 50px;
+`;
+
+const StyledSubmitMessage = styled.p`
+  text-align: center;
+`;
+
+export default function CollectionForm({ onSubmitCollection, prevValues }) {
+  const [confirmMessage, setConfirmMessage] = useState("");
+  const [collectionColor, setCollectionColor] = useState(prevValues?.color);
+  const initialColors = ["#c28375", "#DAF7A6", "#75c297", "#7597c2", "#ab75c2"];
+
+  function handleCancel(event) {
+    event.preventDefault();
+    event.target.form.reset(); // reset the form
+    setCollectionColor(prevValues?.color || ""); // Reset color to previous value
+  }
+
+  return (
+    <>
+      <StyledFormContainer>
+        <StyledForm
+          onSubmit={(e) => {
+            onSubmitCollection(e, prevValues?.id), setConfirmMessage(true);
```

Comment:
instead of putting two function calls into two lines, please separate them with a semicolon
```suggestion
          onSubmit={(e) => {
            onSubmitCollection(e, prevValues?.id);
            setConfirmMessage(true);
```